maven-extension/src/main/java/io/opentelemetry/maven/SpanRegistry.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.maven;
+
+import io.opentelemetry.api.trace.Span;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.component.annotations.Component;
+
+/** Holds the state of the spans in progress FIXME support multi module projects */
```

Comment:
What's needed for multi module (I thought project execution is tied to a module). Just curious

Reply:
I forgot to remove this comment when I added the capability with https://github.com/elastic/opentelemetry-maven-extension/commit/ac37870a32b0bec6ea06066a531ef64d50fd0048 

I primarily had to introduce a "Maven project span" layer between the "Maven build span" and the "Mojo spans".
I removed this outdated comment.